src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -535,6 +537,15 @@ bool Snippet::created() const {
 void Snippet::generate(const jit_snippets_compile_args* jcp) {
     ov::pass::Manager pre_dialect;
     pre_dialect.register_pass<ConvertToSwishCPU>();
+    if (context->getConfig().enforceBF16 &&
+        snippet->has_domain_sensitive_ops() &&
+        dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
```

Comment:
I suggested to check `dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)` when you register nodes in `EnforcePrecision::get_supported_precisions_default`. In general not for all nodes `avx512_core_bf16` architecture is needed to execute in `bf16`. Because of that I suggest to add this check only when you register `Brgemm` in `EnforcePrecision::get_supported_precisions_default` at the moment.

Just imagine. We want to support another `bf16` node, that can be executed in this element type on just `avx512` at least. We register this node in `EnforcePrecision::get_supported_precisions_default` but in reality we don't see that the node is executed in `bf16`. We spend a lot of time to look into the problem. The root cause is we check `dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)` on one level upper in the code for all nodes.

I'm not sure that it's correct behavior. What do you think?

Reply:
1) `dnnl::impl::cpu::x64::mayiuse` should not used in transformation - discussed before
2) tokenization have to provide the correct graph only - discussed before
was removed from condition